langchain/src/vectorstores/rockset.ts

```diff
@@ -0,0 +1,418 @@
+    this.checkIfDestroyed();
+    this.similarityOrder = this.similarityMetric === SimilarityMetric.EuclideanDistance ? "ASC" : "DESC";
+  }
+
+  /**
+   * Embeds and adds Documents to the store.
+   * @param {Documents[]} documents  The documents to store
+   * @returns {Promise<string[]?>}   The _id's of the documents added
+   */
+  async addDocuments(documents: Document[]): Promise<string[] | undefined> {
+    const texts = documents.map(({ pageContent }) => pageContent);
+    return await this.addVectors(
+      await this.embeddings.embedDocuments(texts),
+      documents
+    );
+  }
+
+  /**
+   * Adds vectors to the store given their corresponding Documents
+   * @param {number[][]} vectors   The vectors to store
+   * @param {Document[]} documents The Documents they represent
+   * @return {Promise<string[]?>}  The _id's of the added documents
+   */
+  async addVectors(vectors: number[][], documents: Document[]) {
+    this.checkIfDestroyed();
+    const rocksetDocs = [];
+    for (let i = 0; i < documents.length; i += 1) {
+      const currDoc = documents[i];
+      const currVector = vectors[i];
+      rocksetDocs.push({
+        [this.textKey]: currDoc.pageContent,
+        [this.embeddingKey]: currVector,
+        ...currDoc.metadata
+      })
+    }
+
+    return (await this.client.documents.addDocuments(
+      this.workspaceName, 
+      this.collectionName, {
+        data: rocksetDocs
+      }
+    )).data?.map((docStatus) => docStatus._id || "" )
+  }
+
+  /**
+   * Deletes Rockset documements given their _id's
+   * @param {string[]} ids  The IDS to remove documents with
+   */
+  async delete(ids: string[]): Promise<void> {
+    this.checkIfDestroyed();
+    await this.client.documents.deleteDocuments(
+      this.workspaceName,
+      this.collectionName,
+      {
+        data: ids.map((id) => ({ _id: id }))
+      }
+    )
+  }
+
+  /**
+   * Gets the most relevant documents to a query along
+   * with their similarity score. The returned documents 
+   * are ordered by similarity (most similar at the first
+   * index)
+   * @param {number[]} query  The embedded query to search
+   *                          the store by
+   * @param {number} k        The number of documents to retreive
+   * @param {string?} filter  The SQL `WHERE` clause to filter by
+   */
+  async similaritySearchVectorWithScore(
+    query: number[],
+    k: number,
+    filter?: string
+  ): Promise<[Document, number][]> {
+    this.checkIfDestroyed();
+    if (filter && this.filter) {
+      throw new RocksetStoreError("cannot provide both `filter` and `this.filter`");
+    }
+    const similarityKey = "similarity"
+    const _filter = filter ?? this.filter;
+    return (await this.client.queries.query({
+      sql: {
+        query: `
+          SELECT
+            * EXCEPT("${this.embeddingKey}"),
+            "${this.textKey}",
+            ${this.similarityMetric}([${query}], "${this.embeddingKey}") AS "${similarityKey}"
```

Comment:
Looks like a SQL injection?

Reply:
parameterized query. filter string is tricky, it is something like `age>10` etc - rockset doesn't allow such string params in the API request

Reply:
Ah that's super unfortunate. Oh well.